pyaerocom/utils.py

```diff
@@ -142,3 +144,34 @@ def print_file(path: Path | str):
     for line in path.read_text().splitlines():
         if line.strip():
             print(line)
+
+
+def recursive_defaultdict(d: Mapping | None = None) -> defaultdict:
```

Comment:
Change return-type to `dict`, since that is how it should be used. No need to expose defaultdict.

Recursion can be very dangerous. Would it be possible to limit the recursion-levels on construction, e.g.
```
def recursive_defaultdict(d: Mapping | None = None, max_depth: int) -> defaultdict:
```

Reply:
defaultdict is a subclass of dict, so I think defaultdict is the correct return type. Any check isinstance check for dict will still return true.

recursivedefaultdict only calls the factory function on access of a key in the dictionary, not when the dictionary is initialized. I'm not sure the recursion is a problem in this instance.